pkg/plugin/installer/installer.go

```diff
@@ -36,13 +36,28 @@ type Installer interface {
 	Install() error
 	// Path is the directory of the installed plugin.
 	Path() string
+	// Update updates a plugin to $HELM_HOME.
+	Update() error
 }
 
 // Install installs a plugin to $HELM_HOME.
 func Install(i Installer) error {
+	if _, pathErr := os.Stat(i.Path()); !os.IsNotExist(pathErr) {
+		return errors.New("plugin already exists")
+	}
+
 	return i.Install()
 }
 
+// Update updates a plugin to $HELM_HOME.
```

Comment:
Nit: "updates a plugin in $HELM_HOME" as opposed to "updates a plugin to $HELM_HOME"